.github/workflows/main.yml

```diff
@@ -27,6 +27,75 @@ jobs:
           version: '0.10.0'
       - name: Verify CLI exists
         run: esc version
+  test-individual-keys:
+    strategy: {matrix: {os: [ubuntu-latest]}}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Authenticate with Pulumi Cloud
+        uses: pulumi/auth-actions@v1
+        with:
+          organization: pulumi
+          requested-token-type: urn:pulumi:token-type:access_token:organization
+          cloud-url: https://api.pulumi-staging.io
+      - name: Install ESC and open an environment
+        id: esc
+        uses: ./
+        with:
+          environment: 'pulumi/github/esc-action'
+          keys: 'FOO,SOME_IMPORTANT_KEY'
+          cloud-url: https://api.pulumi-staging.io
+          export-environment-variables: false
+      - name: Verify injection
+        run: |
+          echo "Testing env injection..."
+          REQUIRED_VARS=("K1" "K2")
+          for var in "${REQUIRED_VARS[@]}"; do
+            if [[ -z "${!var}" ]]; then
+              echo "Error: $var is not set or empty" >&2
+              exit 1
+            fi
+            echo "$var is set to: ${!var}"
+          done
+        env:
+          K1: ${{ steps.esc.outputs.FOO }}
+          K2: ${{ steps.esc.outputs.SOME_IMPORTANT_KEY }}
```

Comment:
Can we also test that these variables aren't present if they're not explicitly passed in through `env`?

Reply:
Done